### PR TITLE
3 employes crud

### LINKED ISSUE #3: Employes CRUD
# REQUIREMENTS
Creation of CRUD funcionallity for employee model.

- CREATE
- READ
- UPDATE
- DELETE (Only desactive employee)

### PR DESCRIPTION
**Pull request by Javi1499**
# Description
What did you implemented
- CRUD feutures to employee model
- Delete was implemented as show method
Why did you implemented this:
- Start with elasboration of CRUD after creation model.
- In this case isn't allow delete employee isn't allow and was replaced with deactivate employye in show method

### CODE REVIEW

File: app/views/employees/edit.html.erb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    <div>
+        <label for="position">Position:</label>
+        <input type="text" placeholder="example@example.com" name="position" value="<%= @employee.position %>"/>
+    </div>
+
+    <div>
+        <label for="private_number">Private Number:</label>
+        <input type="password" placeholder="Exam: 123456" name="private_number" value ="<%= @employee.private_number%>"/>
+    </div>
+    <button >Edit Employee</button>
```
Comment (JuanequeX) [inline app/views/employees/edit.html.erb]: Change the identation of this file for 2 spaces

File: app/views/employees/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    
+<%end%>
```
Comment (JuanequeX) [inline app/views/employees/edit.html.erb]: Add a blank lines here

File: app/views/employees/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    <div style="margin-bottom:16px">
+        <h3>
+             <%= employee.id%> <%=employee.name%>
+        </h3>
+        <p>Position: <span><%=employee.position%></span></p>
+
+        <%= link_to 'Edit', edit_employee_path(employee) %>        
+        
+        <%= link_to 'Delete', employee_path(employee) %>
+    </div>
```
Comment (JuanequeX) [inline app/views/employees/index.html.erb]: Same here, with your identation

File: app/views/employees/index.html.erb [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+<%end%>
```
Comment (JuanequeX) [inline app/views/employees/index.html.erb]: Add a blank line here

File: app/views/employees/new.html.erb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        <label for="position">Position:</label>
+        <input type="text" placeholder="example@example.com" name="position"/>
+    </div>
+
+    <div>
+        <label for="private_number">Private Number:</label>
+        <input type="password" placeholder="Exam: 123456" name="private_number"/>
+    </div>
+    <button >Add Employee</button>
+
```
Comment (JuanequeX) [inline app/views/employees/new.html.erb]: Same here with your identation

File: app/views/employees/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        <input type="text" placeholder="example@example.com" name="position"/>
+    </div>
+
+    <div>
+        <label for="private_number">Private Number:</label>
+        <input type="password" placeholder="Exam: 123456" name="private_number"/>
+    </div>
+    <button >Add Employee</button>
+    
+<%end%>
```
Comment (JuanequeX) [inline app/views/employees/new.html.erb]: Add a blank line here

File: config/routes.rb [hunk 7]
```diff
@@ -3,4 +3,6 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
-end
+
+  resources :employees
+end
```
Comment (JuanequeX) [inline config/routes.rb]: Add a blank line here

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/controllers/employees_controller.rb [hunk 8]
```diff
@@ -1,5 +1,30 @@
...
+        employee = Employee.find(params['id'])
+        employee.update name: params['name'], position: params["position"], email:["email"], private_number:["private_number"]
+        redirect_to action: :index
+    end
+
+    def show
+        employee = Employee.find(params['id'])
+        employee.update position: "Desactive"
+        redirect_to action: :index
     end
```
Comment (JuanequeX) [inline app/controllers/employees_controller.rb]: Change the identation of this file for 2 spaces

**Review by JuanequeX — CHANGES_REQUESTED** [review]

**Comment by JuanequeX** [comment]
To add the blank lines in the end of your files automatically they can install this [extension](https://marketplace.visualstudio.com/items?itemName=riccardoNovaglia.missinglineendoffile) in your VS code

File: app/views/employees/edit.html.erb (hunk 1 above)
Comment (Javi1499) [reply app/views/employees/edit.html.erb]: Got it. I'll change this

File: app/views/employees/edit.html.erb (hunk 2 above)
Comment (Javi1499) [reply app/views/employees/edit.html.erb]: Ok. I'll add blank line.

File: app/views/employees/edit.html.erb [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+        <label for="position">Position:</label>
+        <input type="text" placeholder="example@example.com" name="position" value="<%= @employee.position %>"/>
+    </div>
+
+    <div>
+        <label for="private_number">Private Number:</label>
+        <input type="password" placeholder="Exam: 123456" name="private_number" value ="<%= @employee.private_number%>"/>
+    </div>
+    <button >Edit Employee</button>
+
```
Comment (JuanequeX) [inline app/views/employees/edit.html.erb]: Please fix the indentation of this file at 2 spaces

**Review by JuanequeX — CHANGES_REQUESTED** [review]

File: app/helpers/employees_helper.rb [hunk 10]
```diff
@@ -1,2 +1,4 @@
+# frozen_string_literal: true
+
 module EmployeesHelper
 end
```
Comment (YilKanda) [inline app/helpers/employees_helper.rb:4]: If you don't occupy, you can remove it.

File: app/jobs/application_job.rb [hunk 11]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
+
 class ApplicationJob < ActiveJob::Base
   # Automatically retry jobs that encountered a deadlock
   # retry_on ActiveRecord::Deadlocked
```
Comment (YilKanda) [inline app/jobs/application_job.rb]: Remove the code in comments

File: app/views/employees/edit.html.erb (hunk 9 above)
Comment (YilKanda) [inline app/views/employees/edit.html.erb]: I agree with @JuanequeX

File: config/routes.rb [hunk 12]
```diff
@@ -3,4 +3,6 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
```
Comment (YilKanda) [inline config/routes.rb]: Remove the code in comments.

File: app/views/employees/edit.html.erb (hunk 9 above)
Comment (Javi1499) [reply app/views/employees/edit.html.erb]: Done

File: app/helpers/employees_helper.rb (hunk 10 above)
Comment (Javi1499) [reply app/helpers/employees_helper.rb:4]: Deleted it

**Review by JuanequeX — DISMISSED** [review]
LGTM

**Review by femm34 — APPROVED** [review]

**Review by YilKanda — APPROVED** [review]

File: app/controllers/employees_controller.rb [hunk 13]
```diff
@@ -1,5 +1,34 @@
...
+    employee.update name: params['name'], position: params['position'], email: ['email'],
+                    private_number: ['private_number']
+    redirect_to action: :index
+  end
+
+  def show
+    employee = Employee.find(params['id'])
+    employee.update position: 'Desactive'
+    redirect_to action: :index
+  end
```
Comment (vyk2rr) [inline app/controllers/employees_controller.rb:33]: why would you need to set the position to `Desactive` when showing an employee? it doesn't make sense to me when reading the code

Comment (Javi1499) [reply app/controllers/employees_controller.rb:33]: This will change. This part was created to test edit a field. Field that will edit is status.

Comment (EliasUR) [reply app/controllers/employees_controller.rb:33]: This was the way we found to test how it works. We are changing this later when we'll work on the design of this fuction

**Pull request merged**